my_enumerables.rb

```diff
@@ -0,0 +1,121 @@
+module Enumerable
+  def my_all
+    raise 'NO BLOCK GIVEN!' unless block_given?
+
+    arr = to_a
+    arr.my_each_with_index do |_item, index|
+      return false unless yield arr[index]
+    end
+    true
+  end
+
+  def my_any
+    raise 'NO BLOCK GIVEN!' unless block_given?
+
+    arr = to_a
+    index = 0
+    any = false
+
+    arr.my_each do |_item|
+      any = true if yield arr[index]
+
+      index += 1
+    end
+
+    any
+  end
+
+  def my_count(arg = nil)
+    arr = to_a
+
+    return arr.length if arg.nil?
+
+    index = 0
+    count = 0
+
+    arr.my_each do |_item|
+      count += 1 if arg == arr[index]
+      index += 1
+    end
+
+    count
+  end
+
+  def my_each
+    arr = to_a
+    i = 0
+    while i < arr.length
+      yield arr[i]
+      i += 1
+    end
+
+    arr
+  end
+
+  def my_each_with_index
+    arr = to_a
+    i = 0
+    while i < arr.length
+      yield arr[i], i
+      i += 1
+    end
+
+    arr
+  end
+
+  def my_inject
+    raise 'ERROR' unless block_given?
+
+    arr = to_a
+    i = 1
+    result = arr[0]
+    result = '' if result.class == String
+
+    while i < arr.length
+      result = yield result, arr[i]
+      i += 1
+    end
+
+    result
```

Comment:
- [ ] #my_inject raises a "LocalJumpError" when no block or argument is given
      

```Ruby
array.my_inject
      
expected LocalJumpError, got #<RuntimeError: ERROR> with backtrace:
```

- [ ]  the method doesn't work when giving it an argument, please solve that and make it work with all scenarios that need an argument
here's a list of what's not working: 

    - [ ] #my_inject when a block is given with an initial value combines all elements of enum by applying a binary operation, specified by a block
    - [ ] #my_inject when a symbol is specified without an initial value combines each element of the collection by applying the symbol as a named method
    - [ ] #my_inject when a symbol is specified without an initial value combines each element of the collection by applying the symbol as a named method:range
    - [ ] #my_inject when a symbol is specified with an initial value combines each element of the collection by applying the symbol as a named method
    - [ ] #my_inject when a symbol is specified with an initial value combines each element of the collection by applying the symbol as a named method::range